common-arcana.lic

```diff
@@ -236,4 +236,57 @@ module DRCA
       return parse_mana_message(mana_msg)
     end
   end
+
+  def check_osrel(osrel_no_harness, osrel_amount)
+    return unless DRSpells.active_spells['Osrel Meraud'] && DRSpells.active_spells['Osrel Meraud'] < 90
+    infuse_om(!osrel_no_harness, osrel_amount)
+  end
+
+  def do_waggle_set(settings, spells)
+    check_osrel(settings.osrel_no_harness, settings.osrel_amount)
+    spells.each do |name, data|
+      next if DRSpells.active_spells[name] && (data['recast'].nil? || DRSpells.active_spells[name].to_i > data['recast'])
+      while mana < 40
+        echo('Waiting on mana...')
+        pause 15
+      end
+
+      if data['ritual']
+        ritual(data)
+      else
+        cast_spell(settings, data)
+      end
+    end
+  end
+
+
+  def cast_spell(settings, data)
```

Comment:
I waffled between having the settings be an argument or having the method call get_settings.